### ISSUE TITLE
Things to consider during final review

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#403]
The Preferences, Toolbar, and John Travoltage all sound awesome. Everything in issue #401 has been addressed.

I have some concerns which we may or may not want to address before first publication. Not listing in any order, just how I found them:

**Possible Screen Reader Accessibility Issues:**
- [ ] (1) When Voicing is off and I am using VoiceOver, I have to arrow through the list items representing the tabs before entering the panel content of the tab. For example, when on the General Tab, I have to arrow through the bulleted Audio, Visual Tabs before getting the General Tab H2 heading and paragraphs. 
    - Not sure of solution, but wondering if we are using same-page links for the tab list items and linking them to their sections (tab panels). Can we create mark-up that allows the reader to go directly from the Tab to the content in the Tab Panel?
- [x] (2) When Voicing is off and I am using VoiceOver, the reading blocks in the Preferences menu show their role which is a button. This may be an issue for non-visual users as they use a screen reader to set their preferences. They may try, unsuccessfully, to activate these non-working reading block buttons.
    - Not sure of solution, but we may need to look at a more robust custom solution where the reading block content is marked up in a p-element until voicing is enabled, and then the reading blocks become buttons. I have no idea of this is possible.
- [x] (3) Not hearing context responses for switches and checkboxes when using a screen reader.
 
**Minor Voicing Issues:**
- [x] (4) The pronunciation of "indirect" is not great. It sounds awkward in the voicing-context-response for the "context responses" checkbox. We could consider adjusting the wording a little more. I put some ideas in the design doc, but we could just eliminate "indirect" in the response and use the same wording that is in the checkbox label:
    - Voicing **other sim** changes.
    - **Other sim** changes muted. 
    - opened in issue in joist for to change context responses https://github.com/phetsims/joist/issues/712

**Comment by terracoda** [comment]
@jessegreenberg, I think you could go ahead and implement that minor Voicing issue, and the rest we can discuss at next meeting and determine if we fix now or wait until version 2 of Preferences.

**Comment by terracoda** [comment]
@jessegreenberg, I think you mentioned that number (2) is a bug. There should be no buttons when voicing on the help text when voicing is off. Let me know when I can verify a new version.

**Comment by jessegreenberg** [comment]
These things have been addressed @terracoda. The only thing I did not make changes for was (1). I tested with VoiceOver in Safari and NVDA in Firefox and both screen readers switched to forms mode for the Preferences tabs, so I was able to cycle through them with arrow keys. Let me know if this is still a problem for you and I can investigate further. Can you please verify the other things in this issue have been resolved with this link?
https://phet-dev.colorado.edu/html/john-travoltage/1.6.0-preferences.6/phet/john-travoltage_en_phet.html

**Comment by terracoda** [comment]
@jessegreenberg, entering the Visual Tab and the Audio Tab works great because those tabs have focusable items in them. For the General Tab there is no focusable item, so the Tab key takes my focus to Safari's Go back button.
 It works great when voicing is enabled because the paragraphs are focusable reading blocks.

The down arrow doesn't do anything. I think the down arrow should move focus to the Tab Panel itself.

In the Pickering article it says, "we can intercept the down arrow key press and move focus programmatically to the open panel itself. See panels[i].focus() in the following snippet:".

Here's the JS code snippet:
```javascript
tab.addEventListener('keydown', e => {  
  let index = Array.prototype.indexOf.call(tabs, e.currentTarget);
  let dir = e.which === 37 ? index - 1 : e.which === 39 ? index + 1 : e.which === 40 ? 'down' : null;
  if (dir !== null) {
    e.preventDefault();
    dir === 'down' ? panels[i].focus() : tabs[dir] ? switchTab(e.currentTarget, tabs[dir]) : void 0;
  }
});
```

Is this down-arrow interception implemented?

**Comment by jessegreenberg** [comment]
> Is this down-arrow interception implemented?

Yes, but it is scripted to move focus between tabs just like the left/right arrow keys. I think we can change it, it will just require some changes to the way the tabs and panels have been implemented. I would guess ~90 minutes to implement, test, verify.

However, I am not a big fan of [Pickerings design here](https://inclusive-components.design/tabbed-interfaces/). Once you arrow in to the tab panel you can't arrow back out! And I agree with his statement about focusaing non-interactive things
> The general rule is not to allow the focus of non-interactive elements by the user because the expectation is that focusable elements will each actually do something.

But let me know if you would like me to proceed with this @terracoda and Ill make the change.

**Comment by terracoda** [comment]
Hmm, I think the design is quite innovative and the flow is very natural. 

I agree we are stepping outside the box of what things are considered for receiving focus, but if you think about the the tab-panels somewhat like a second-level option in combobox with a top-level options that lead to a second level of options, it kind of makes sense. The tab-panel is at the second level. The tab-panels are part of the interactive interface. The paragraphs held within the tab-panels are not interactive and do not actually receive focus. Not being able differentiate between the paragraphs and the tab-panel that contains them is what I think feels strange. 

It seems stranger to me to NOT be able to easily start reading from the start of the tab-panel. Using the down arrow to activate the tab-panel allows me to easily move into the content of the panel when that content itself is not directly interactive.

**Comment by terracoda** [comment]
@jessegreenberg, please implement it.

**Comment by terracoda** [comment]
@jessegreenberg, my apologies. I forgot to re-assign to you.

**Comment by jessegreenberg** [comment]
I had forgotten about this issue, creating a new one for this chunk of work.
